bug(printer): nil pointer panic in HTML and JUnit printers when resource is missing from AllResources

```markdown
## Description
If a resource exists in a failed control's list of resources, but is missing from `opaSessionObj.AllResources` (which can happen on transient network issues, filtered/pruned resource lists, or custom mock outputs), both the HTML and JUnit printers will crash the Kubescape CLI with a nil pointer panic.

While `resourcetable.go` guards against missing resources:
```go
resource, ok := opaSessionObj.AllResources[resourceID]
if !ok {
    continue
}
```

Neither `htmlprinter.go` nor `junit.go` has this check:
1. In `htmlprinter.go`, `buildResourceTableView` gets the resource and appends it to `ResourceResult` even if it is `nil`. When sorting by namespace, it calls `.Resource.GetNamespace()`, resulting in a panic.
2. In `junit.go`, `testsCases` retrieves `resource := results.AllResources[rId]` and immediately passes it to `resourceToString`, which calls `resource.GetApiVersion()`, triggering a panic.

---

## Steps To Reproduce / Stack Trace
Running a mock scan session where a control has a failed resource ID `r-1` that does not exist in `AllResources` triggers the panic:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x18]

goroutine 85 [running]:
github.com/kubescape/kubescape/v3/core/pkg/resultshandling/printer/v2.resourceToString({0x0, 0x0}, {0x0, 0x0})
	/core/pkg/resultshandling/printer/v2/junit.go:291 +0x28
github.com/kubescape/kubescape/v3/core/pkg/resultshandling/printer/v2.testsCases(0x526f76a6fd50, {0x10bdc56f8, 0x526f76a68078}, {0x107a34aa6, 0x9})
	/core/pkg/resultshandling/printer/v2/junit.go:248 +0x5c4
github.com/kubescape/kubescape/v3/core/pkg/resultshandling/printer/v2.listTestsSuite(0x526f76a6fd50)
	/core/pkg/resultshandling/printer/v2/junit.go:197 +0x314
github.com/kubescape/kubescape/v3/core/pkg/resultshandling/printer/v2.testsSuites(0x526f759b0b28?)
	/core/pkg/resultshandling/printer/v2/junit.go:173 +0x24
github.com/kubescape/kubescape/v3/core/pkg/resultshandling/printer/v2.(*JunitPrinter).ActionPrint(0x526f76a6fd40, {0x10bdc5688, 0x10c42ad28}, 0x526f759b0f48?, {0x1?, 0x104a7c980?, 0x526f759b0c38?})
	/core/pkg/resultshandling/printer/v2/junit.go:134 +0x34
```

---

## Suggested Fix
1. In `htmlprinter.go`, verify the resource exists in `AllResources` before appending it to `ResourceResult`.
2. In `junit.go`, guard against `resource == nil` inside `testsCases` or gracefully print fallback info in `resourceToString`.
```

--- a/core/cautils/datastructures.go
+++ b/core/cautils/datastructures.go
@@ -9,6 +9,8 @@ import (
 	"github.com/anchore/grype/grype/vulnerability"
 	"github.com/anchore/syft/syft/sbom"
 	"github.com/armosec/armoapi-go/armotypes"
+	"github.com/kubescape/go-logger"
+	"github.com/kubescape/go-logger/helpers"
 	"github.com/kubescape/k8s-interface/workloadinterface"
 	"github.com/kubescape/opa-utils/reporthandling"
 	apis "github.com/kubescape/opa-utils/reporthandling/apis"
@@ -130,15 +132,21 @@ func (sessionObj *OPASessionObj) SetTopWorkloads() {
 
 	// set top workloads according to number of top workloads
 	topWorkloads := make([]reporthandling.IResource, 0, TopWorkloadsNumber)
-	for i := range TopWorkloadsNumber {
-		if i >= len(topWorkloadsSorted) {
+	for _, wl := range topWorkloadsSorted {
+		if len(topWorkloads) >= TopWorkloadsNumber {
 			break
 		}
 
-		source := sessionObj.ResourceSource[topWorkloadsSorted[i].ResourceID]
+		source := sessionObj.ResourceSource[wl.ResourceID]
 
+		res, ok := sessionObj.AllResources[wl.ResourceID]
+		if !ok {
+			logger.L().Debug("resource missing from AllResources, skipping",
+				helpers.String("resourceID", wl.ResourceID))
+			continue
+		}
 		wlObj := &reporthandling.Resource{
-			IMetadata: sessionObj.AllResources[topWorkloadsSorted[i].ResourceID],
+			IMetadata: res,
 			Source:    &source,
 		}
 

--- a/core/cautils/datastructures_test.go
+++ b/core/cautils/datastructures_test.go
@@ -4,6 +4,7 @@ import (
 	"context"
 	"testing"
 
+	"github.com/kubescape/k8s-interface/workloadinterface"
 	"github.com/kubescape/opa-utils/reporthandling"
 	"github.com/kubescape/opa-utils/reporthandling/results/v1/prioritization"
 	reporthandlingv2 "github.com/kubescape/opa-utils/reporthandling/v2"
@@ -124,6 +125,9 @@ func TestSetTopWorkloads(t *testing.T) {
 		obj.ResourcesPrioritized = map[string]prioritization.PrioritizedResource{
 			"res1": {ResourceID: "res1", Score: 80},
 		}
+		obj.AllResources = map[string]workloadinterface.IMetadata{
+			"res1": workloadinterface.NewWorkloadObj(map[string]any{}),
+		}
 		obj.ResourceSource = map[string]reporthandling.Source{}
 
 		obj.SetTopWorkloads()
@@ -136,6 +140,9 @@ func TestSetTopWorkloads(t *testing.T) {
 		obj.Report = nil
 		obj.ResourcesPrioritized = map[string]prioritization.PrioritizedResource{
 			"res1": {ResourceID: "res1", Score: 50},
+		}
+		obj.AllResources = map[string]workloadinterface.IMetadata{
+			"res1": workloadinterface.NewWorkloadObj(map[string]any{}),
 		}
 		obj.ResourceSource = map[string]reporthandling.Source{}
 
@@ -151,6 +158,10 @@ func TestSetTopWorkloads_Idempotent(t *testing.T) {
 	obj.ResourcesPrioritized = map[string]prioritization.PrioritizedResource{
 		"res1": {ResourceID: "res1", Score: 100},
 		"res2": {ResourceID: "res2", Score: 90},
+	}
+	obj.AllResources = map[string]workloadinterface.IMetadata{
+		"res1": workloadinterface.NewWorkloadObj(map[string]any{}),
+		"res2": workloadinterface.NewWorkloadObj(map[string]any{}),
 	}
 
 	obj.ResourceSource = map[string]reporthandling.Source{}

--- a/core/pkg/resultshandling/printer/v2/attacktracks.go
+++ b/core/pkg/resultshandling/printer/v2/attacktracks.go
@@ -6,6 +6,8 @@ import (
 	"strings"
 
 	"github.com/jwalton/gchalk"
+	"github.com/kubescape/go-logger"
+	"github.com/kubescape/go-logger/helpers"
 	"github.com/kubescape/kubescape/v3/core/cautils"
 	"github.com/kubescape/kubescape/v3/core/pkg/resultshandling/gotree"
 	"github.com/kubescape/opa-utils/reporthandling/apis"
@@ -94,18 +96,27 @@ func (prettyPrinter *PrettyPrinter) printAttackTracks(opaSessionObj *cautils.OPA
 		return resources[i].Score > resources[j].Score
 	})
 
-	for i := 0; i < topResourceCount && i < len(resources); i++ {
-		fmt.Fprintf(prettyPrinter.writer, "\n%s\n", getSeparator("^"))
-		resource := resources[i]
-		resourceObj := opaSessionObj.AllResources[resource.ResourceID]
+	printedCount := 0
+	for _, resource := range resources {
+		if printedCount >= topResourceCount {
+			break
+		}
 
+		resourceObj, ok := opaSessionObj.AllResources[resource.ResourceID]
+		if !ok {
+			logger.L().Debug("resource missing from AllResources, skipping",
+				helpers.String("resourceID", resource.ResourceID))
+			continue
+		}
+
+		fmt.Fprintf(prettyPrinter.writer, "\n%s\n", getSeparator("^"))
 		fmt.Fprintf(prettyPrinter.writer, "Name: %s\n", resourceObj.GetName())
 		fmt.Fprintf(prettyPrinter.writer, "Kind: %s\n", resourceObj.GetKind())
 		fmt.Fprintf(prettyPrinter.writer, "Namespace: %s\n\n", resourceObj.GetNamespace())
 
 		fmt.Fprintf(prettyPrinter.writer, "Score: %.2f\n", resource.Score)
 		fmt.Fprintf(prettyPrinter.writer, "Severity: %s\n", apis.SeverityNumberToString(resource.Severity))
-		fmt.Fprintf(prettyPrinter.writer, "Total vectors: %v\n\n", len(resources[i].PriorityVector))
+		fmt.Fprintf(prettyPrinter.writer, "Total vectors: %v\n\n", len(resource.PriorityVector))
 
 		if v, found := resourceToAttackTrack[resource.ResourceID]; found {
 			prettyPrinter.printResourceAttackGraph(v)
@@ -115,7 +126,7 @@ func (prettyPrinter *PrettyPrinter) printAttackTracks(opaSessionObj *cautils.OPA
 			return resource.PriorityVector[x].Score > resource.PriorityVector[y].Score
 		})
 
-		for j := 0; j < topVectorCount && j < len(resources[i].PriorityVector); j++ {
+		for j := 0; j < topVectorCount && j < len(resource.PriorityVector); j++ {
 			priorityVector := resource.PriorityVector[j]
 
 			vectorStrings := []string{}
@@ -125,5 +136,6 @@ func (prettyPrinter *PrettyPrinter) printAttackTracks(opaSessionObj *cautils.OPA
 
 			fmt.Fprintf(prettyPrinter.writer, "%v) [%.2f] [Severity: %v] [Attack Track: %v]: %v \n", j+1, priorityVector.Score, apis.SeverityNumberToString(priorityVector.Severity), priorityVector.AttackTrackName, strings.Join(vectorStrings, " -> "))
 		}
+		printedCount++
 	}
 }

--- a/core/pkg/resultshandling/printer/v2/htmlprinter.go
+++ b/core/pkg/resultshandling/printer/v2/htmlprinter.go
@@ -136,7 +136,12 @@ func buildResourceTableView(opaSessionObj *cautils.OPASessionObj) ResourceTableV
 	resourceTableView := make(ResourceTableView, 0)
 	for resourceID, result := range opaSessionObj.ResourcesResult {
 		if result.GetStatus(nil).IsFailed() {
-			resource := opaSessionObj.AllResources[resourceID]
+			resource, ok := opaSessionObj.AllResources[resourceID]
+			if !ok {
+				logger.L().Debug("resource missing from AllResources, skipping",
+					helpers.String("resourceID", resourceID))
+				continue
+			}
 			ctlResults := buildResourceControlResultTable(result.AssociatedControls, &opaSessionObj.Report.SummaryDetails)
 			resourceTableView = append(resourceTableView, ResourceResult{resource, ctlResults})
 		}

--- a/core/pkg/resultshandling/printer/v2/htmlprinter_test.go
+++ b/core/pkg/resultshandling/printer/v2/htmlprinter_test.go
@@ -5,6 +5,7 @@ import (
 	"os"
 	"testing"
 
+	"github.com/kubescape/kubescape/v3/core/cautils"
 	"github.com/kubescape/opa-utils/reporthandling/apis"
 	"github.com/kubescape/opa-utils/reporthandling/results/v1/reportsummary"
 	"github.com/kubescape/opa-utils/reporthandling/results/v1/resourcesresults"
@@ -88,4 +89,23 @@ func TestBuildResourceControlResultTable_MissingControl(t *testing.T) {
 		results := buildResourceControlResultTable([]resourcesresults.ResourceAssociatedControl{ac}, summaryDetails)
 		assert.Empty(t, results)
 	})
+}
+
+func TestBuildResourceTableView_SkipsMissingResource(t *testing.T) {
+	session := cautils.NewOPASessionObjMock()
+	session.ResourcesResult = map[string]resourcesresults.Result{
+		"r-1": {
+			ResourceID: "r-1",
+			AssociatedControls: []resourcesresults.ResourceAssociatedControl{
+				{
+					ControlID: "C-0001",
+					Status:    apis.StatusInfo{InnerStatus: apis.StatusFailed},
+				},
+			},
+		},
+	}
+	// Do not populate session.AllResources["r-1"]
+
+	view := buildResourceTableView(session)
+	assert.Empty(t, view, "missing resource should be skipped, not included with nil")
 }

--- a/core/pkg/resultshandling/printer/v2/junit.go
+++ b/core/pkg/resultshandling/printer/v2/junit.go
@@ -240,7 +240,13 @@ func testsCases(results *cautils.OPASessionObj, controls reportsummary.IControls
 					continue
 				}
 
-				resource := results.AllResources[rId]
+				resource, ok := results.AllResources[rId]
+				if !ok {
+					logger.L().Debug("resource missing from AllResources, reporting by ID",
+						helpers.String("resourceID", rId))
+					resources[fmt.Sprintf("resourceID: %s", rId)] = nil
+					continue
+				}
 				sourcePath := ""
 				if ResourceSourcePath, ok := results.ResourceSource[rId]; ok {
 					sourcePath = ResourceSourcePath.RelativePath

--- a/core/pkg/resultshandling/printer/v2/junit_test.go
+++ b/core/pkg/resultshandling/printer/v2/junit_test.go
@@ -15,6 +15,7 @@ import (
 
 	"github.com/kubescape/kubescape/v3/core/cautils"
 	"github.com/kubescape/opa-utils/reporthandling/apis"
+	helpersv1 "github.com/kubescape/opa-utils/reporthandling/helpers/v1"
 	"github.com/kubescape/opa-utils/reporthandling/results/v1/reportsummary"
 	reporthandlingv2 "github.com/kubescape/opa-utils/reporthandling/v2"
 	"github.com/stretchr/testify/assert"
@@ -737,4 +738,42 @@ func TestJunitActionPrintComplianceScore(t *testing.T) {
 				"complianceScore must come from ComplianceScore (%s), not Score", tt.want)
 		})
 	}
+}
+
+func TestJunitActionPrintMissingResourceNoPanic(t *testing.T) {
+	session := cautils.NewOPASessionObjMock()
+
+	resourceIDs := helpersv1.AllLists{}
+	resourceIDs.Append(apis.StatusFailed, "r-1")
+
+	control := reportsummary.ControlSummary{
+		ControlID: "C-0001",
+		Name:      "Test Control",
+		Status:    apis.StatusFailed,
+		StatusInfo: apis.StatusInfo{
+			InnerStatus: apis.StatusFailed,
+		},
+		ResourceIDs: resourceIDs,
+	}
+
+	session.Report = &reporthandlingv2.PostureReport{
+		SummaryDetails: reportsummary.SummaryDetails{
+			Controls: reportsummary.ControlSummaries{
+				"C-0001": control,
+			},
+		},
+	}
+
+	tmp, err := os.CreateTemp("", "junit-test-*.xml")
+	require.NoError(t, err)
+	defer os.Remove(tmp.Name())
+
+	jp := NewJunitPrinter(false)
+	jp.writer = tmp
+
+	assert.NotPanics(t, func() {
+		jp.ActionPrint(context.Background(), session, nil)
+	})
+
+	require.NoError(t, tmp.Close())
 }

